SIP-028: sBTC Signer Criteria (#186)
stacksgov/sips: sips/sip-026/sip-026-sbtc_peg.md

# sBTC SIP

> ## SIP Number: 022

Reviewer: ```suggestion
## SIP Number: 028
```

Reviewer: Additionally, the path to the files should be modifed from `sip-026`->`sip-028` per @Hero-Gamer

**Title:** A Decentralized Two-Way Bitcoin Peg  
**Type:** Standard  
**Status:** Draft  

## Abstract
sBTC is a novel digital asset that lets you move Bitcoin in and out of the Stacks blockchain. With sBTC, users can interact with Clarity smart contracts, which enable Bitcoin applications such as payments, decentralized lending, decentralized exchanges, and BTC-backed stablecoins.

> sBTC is a SIP-010 token on the Stacks blockchain, backed 1:1 against BTC, and operated by a decentralized set of signers. When BTC is locked on the Bitcoin L1, an equivalent amount of sBTC is issued on the Stacks layer, ensuring a consistent 1:1 ratio of sBTC:BTC. Users can redeem their sBTC at any time by submitting a withdrawal request. Once the request is processed by sBTC signers, BTC is returned to the user’s specified address on the Bitcoin L1.

Reviewer: ```suggestion
sBTC is a [SIP-010](https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md) token on the Stacks blockchain, backed 1:1 against BTC, and operated by a decentralized set of signers. When BTC is locked on the Bitcoin L1, an equivalent amount of sBTC is issued on the Stacks layer, ensuring a consistent 1:1 ratio of sBTC:BTC. Users can redeem their sBTC at any time by submitting a withdrawal request, and once the request is processed by the sBTC signers, BTC is returned to the user’s specified address on the Bitcoin L1.
```

> sBTC will have a crucial role in scaling Bitcoin, as well as introducing new and innovative functionalities to users, growing the size of the overall Bitcoin ecosystem. This SIP aims to describe the sBTC system, the process for signer selection, and the features available in the initial release compared to subsequent versions of sBTC. It does not attempt to describe the low-level technical details of any subsequent release, which will be provided in a future SIP.

Reviewer: I'd prefer to not use the word scaling here if possible, because that would require a lot of explanation about what it is that is meant.
Alternatively, if you feel strongly about keeping that wording - maybe a short explanation of what that means in the context of sbtc (possible in contrast to lightning?)

## Introduction

## Glossary

| Term                     | Definition |
|--------------------------|------------|

> | SIP-10 Token             | A token on the Stacks blockchain that adheres to the fungible token standards outlined in SIP-10. |

Reviewer: ```suggestion
| SIP-10 Token             | A token on the Stacks blockchain that adheres to the fungible token standards outlined in [SIP-10](https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md). |
```

| sBTC                     | A SIP-10 token on the Stacks Blockchain that can be turned back into BTC on the Bitcoin Blockchain. 1 sBTC is equivalent to 1 BTC on the Bitcoin Blockchain. |
| sBTC operation           | An operation that initiates some action from the sBTC protocol. |
| Bitcoin holder           | A person or system holding BTC, interested in acquiring sBTC. |

> | sBTC holder              | A person or system holding sBTC, interested in turning it back into BTC. |

Reviewer: ```suggestion
| sBTC holder              | A person or system holding sBTC, interested in converting back into BTC. |
```

- **Ecosystem Alignment:** Commitment to the growth and principles of the Stacks and Bitcoin ecosystems with contributions to advance the network.

By implementing a whitelisted signer set for the sBTC Bootstrap release, this helps to reduce time to market to launch sBTC sooner, remove complexity from the initial protocol design, and ensure the sBTC protocol is secure at launch.

### Activation
sBTC Bootstrap is designed to activate on Stacks Nakamoto as defined in SIP-021. Therefore, this SIP is only meaningful when SIP-021 activates. The sBTC Working Group plans to observe at least 2-4 weeks of network behavior on Stacks Nakamoto to ensure a stable release. After this period, sBTC Bootstrap can be activated on the Stacks network without requiring a separate hard fork.

**Process of Activation**
Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The criteria for the stacker and non-stacker voting is as follows.

**For Stackers:**
In order for this SIP to activate, the following criteria must be met by the set of Stacked STX:

- At least 80 million Stacked STX must vote at all to activate this SIP.

> - Of the Stacked STX that vote, at least 80% of them must vote "yes."

Reviewer: Can you briefly explain why this specific criteria? Is there historical precedence, or some other reason these values were chosen?

> # SIP-028: sBTC Bootstrap

Reviewer 2: ```suggestion
# Preamble
```

- **SIP Number:** 028

> - **Title:** A Decentralized and Programmable Asset Backed 1:1 with BTC 

Reviewer 3: This SIP is supposed to be describing the _criteria for signers_ for such a system, not the system itself.  The title should reflect this.

- **Consideration:** Technical, Economics

> - **Type:** Standard

Reviewer 4: It seems that this may just be Informational now?

- **Status:** Draft

> - **Authors:**

Reviewer 3: Please alphabetize and include email addresses

> - **SIP Number:** 028
> - **Title:** A Decentralized and Programmable Asset Backed 1:1 with BTC 
> - **Consideration:** Technical, Economics
> - **Type:** Informational
> - **Status:** Draft
> - **Authors:**
>   - Andre Serrano
>   - Ashton Stephens
>   - Joey Yandle
>   - Mårten Blankfors
>   - Jesus Najera
>   - Jude Nelson
>   - Friedger Müffke
>   - Tycho Onnasch

Reviewer 2: ```suggestion
SIP Number: 028

Title: sBTC Bootstrap - A Decentralized and Programmable Asset Backed 1:1 with BTC 

Authors:
    Andre Serrano
    Ashton Stephens
    Joey Yandle
    Mårten Blankfors
    Jesus Najera
    Jude Nelson
    Friedger Müffke
    Tycho Onnasch

Consideration: Technical, Economics

Type: Informational

Status: Draft

Discussions-To: <link to forum / workgroup discussions???>
```

Reviewer 2: Should `Type: Operation` ? since this describes standard for sBTC signer node operators? 

another option is `Type: Standard` with `Layer: Application` ?

## Abstract
sBTC is a SIP-010 token on the Stacks blockchain, backed 1:1 against BTC, and operated by a decentralized set of signers such that only a 70% majority can access the funds to maintain the protocol.

> While Bitcoin on its own cannot be used with smart contracts, sBTC provides a bridge of value from the Bitcoin Blockchain to the Stacks Blockchain to enable users who only own BTC to utilize the capability of smart contracts without switching to a new currency. This SIP aims to describe the high level sBTC system and the criteria for signer selection. 

Reviewer 3: Typo: `Bitcoin Blockchain` --> `Bitcoin blockchain` (capitalization)

Reviewer 3: "enable users who only own BTC to utilize the capability of smart contracts without switching to a new currency"

I think you need to make a choice.  Either this SIP proposes using sBTC as a gas token (in which case, a full-blown technical description of warranted),  or this SIP describes an auxiliary service that _could_ be built later with sBTC but is not part of this SIP.  If you want the latter, then this allusion should be dropped.

Reviewer 3: Actually, ignore this.  Just use the abstract I have below.

> This first iteration of sBTC is purely at the application layer and does not affect network consensus or operation. This SIP does not attempt to describe the technical details of this or any subsequent sBTC releases; it only seeks to get community approval for the requirements of the sBTC bootstrap signers.

Reviewer 3: The abstract still reads like the authors can't decide of this SIP is going to be a technical SIP describing a particular contract, or a SIP that only focuses on soliciting signers for a forthcoming system.

It sounds like the latter is the goal, and not the former?  If so, then I think this abstract needs to be rewritten.

> ## Abstract

> This SIP takes the position that Stacks can play a key role in offering a rich programming environment for Bitcoin with low-latency transactions.  This would be achieved with a new wrapped Bitcoin asset, called sBTC, which would be implemented on Stacks 3.0 and later as a SIP-010 token.  Stacks today offers a smart contract runtime for Stacks-hosted assets, and the forthcoming Stacks 3.0 release [link to SIP-021] provides lower transaction latency than Bitcoin for Stacks transactions.  By providing a robust BTC-wrapping mechanism based on threshold signatures, users would be able to lock their real BTC on the Bitcoin chain, instantiate an equal amount of sBTC tokens on Stacks, use these sBTC tokens on Stacks, and eventually redeem them for real BTC at 1:1 parity, minus the cost of the relevant blockchain transaction fees.

> This is the first of several SIPs that describe such a system.  This SIP describes the threshold signature mechanism, and solicits from the ecosystem both a list of signers and the criteria for vetting them.  These sBTC signers would be responsible for collectively holding all locked BTC, and redeeming sBTC for BTC upon request.  Given the high-stakes nature of their work, the authors of this SIP believe that such a wrapped asset can only be made to work in practice if the Stacks ecosystem members can reach broad consensus on how these signers are chosen.  Thus, the first sBTC SIP put forth for activation concerns the selection of sBTC signers.

> This SIP outlines but does not describe in technical detail the workings of the first sBTC system.  A separate SIP will be written to do so if this SIP successfully activates.

> The sBTC Bootstrap phase is part of an iterative release process to simplify implementation and accelerate the sBTC release timeline. The initial phase does not include the complete feature set described in the [original sBTC design documents](https://github.com/stacksgov/sips/blob/e6b3233e76c22cfd6ef02f21add66696b9e4c314/sips/sip-025/sip-025-sbtc.md). This SIP does not attempt to describe the low-level technical details of any subsequent releases, which will be provided in a future SIP addendum. 

Reviewer 4: Let's make it very clear here in the abstract what this SIP is defining. How about changing that last sentence to something like this?

This first iteration of sBTC is purely at the application layer and does not affect network consensus or operation. This SIP does not attempt to describe the technical details of this or any subsequent sBTC releases; it only seeks to get community approval for the requirements of the sBTC bootstrap signers.

## Introduction

### Glossary

| Term               | Definition                                                                                              |
|--------------------|---------------------------------------------------------------------------------------------------------|
| SIP-10 Token       | A token on the Stacks blockchain that adheres to the fungible token standards outlined in [SIP-10](https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md).        |

> | sBTC               | A SIP-10 token on the Stacks Blockchain that can be turned back into BTC on the Bitcoin Blockchain. 1 sBTC is equivalent to 1 BTC on the Bitcoin Blockchain. |

Reviewer 3: Pervasively, "Bitcoin Blockchain" is a typo.  "Blockchain" is not capitalized here.

| sBTC operation     | An operation that initiates some action from the sBTC protocol.                                         |
| .sbtc contract     | A smart contract (or a collection of contracts) defining the sBTC token and functions related to it     |

> | sBTC Peg Wallet    | The single UTXO holding the entire BTC balance that’s pegged into sBTC. This peg wallet is managed and maintained by the sBTC Signers. |

Reviewer 3: Fluency: no contractions, please.  Use "that is" instead of "that's"

| Stacks Signer      | An entity that receives PoX payouts for stacking their STX tokens and actively participating in the Stacks protocol by signing mined blocks. |
| sBTC Signer        | An entity that will sign sBTC operations and communicate with contracts on the chain to make that feasible. This entity has partial access to spending the sBTC UTXO. In this release, the sBTC signer is wholly separate from the Stacks Signer |

> | sBTC Signer Set    | The set of all sBTC signers. Each is registered with the .sbtc contract and the transfer. These entities as a group have full democratic access to the sBTC UTXO. |

Reviewer 3: Reword as "These entities as a group collectively maintain the sBTC's Bitcoin UTXO."  "full democratic access" is prescriptive of a technical implementation that may not even come to pass.

Reviewer: [95a12a1](https://github.com/andrerserrano/sips/commit/95a12a1514bcdb3564a3e4e1cab8b35402f6d3b2)

> | sBTC Signer API    | API exposed by the sBTC Signer that handles basic low level commands. Remains private except to the Deposit API. |

Reviewer 3: Please use full sentences here, since you have done so above.

| Deposit API        | A third party API that communicates with the sBTC Bootstrap Signers via the sBTC Signer API.             |

## Problem Statement

> Bitcoin is limited in its programmability and scalability. While its security and censorship-resistance make it a compelling platform to build decentralized applications, transaction confirmation times do not meet the modern needs of users and developers. 

Reviewer 3: Replace "scalability" with either "slowness"  or "high latency"

- **Bitcoin’s Limited Programmability:** Bitcoin’s script has limited programmability, making it unsuitable for most decentralized applications. This leaves users with very few options to use financial applications, like lending and trading, without entrusting their Bitcoin to centralized entities. This also exposes users to counterparty risk, which has the potential to result in lost funds. 

> - **Bitcoin’s Slow Transaction Times:** Bitcoin is limited in its ability to process large amounts of data quickly and efficiently. Today, Bitcoin creates new blocks every 10 minutes on average, an interval which is longer than some alternative blockchains. This prohibits many types of applications that require faster confirmation times. 

Reviewer 3: Can you give examples of applications that need faster confirmations?

Also, I think it's good enough to just say that Bitcoin's slow transaction times _and_ their slow confirmation times are a well-known and widely-felt source of user frustration, so _anything_ we can do to speed up transaction confirmation would be a win.

## Proposed Solution

> sBTC aims to solve Bitcoin’s limitations by combining the capability of the Stacks Blockchain with the stability of Bitcoin’s value. By enabling secure movement of BTC in and out of the Stacks Blockchain via the sBTC protocol, users can interact with their BTC on Stacks using Clarity smart contracts and fast block times. Users can deposit BTC into the protocol, seamlessly transact using sBTC on the Stacks blockchain and have the freedom to redeem sBTC tokens for the underlying BTC at any time. 

Reviewer 3: * sBTC doesn't solve Bitcoin's limitations; it _addresses_ them.

* Bitcoin's value isn't "stable"

* "By enabling [the] secure movement of Bitcoin" <-- typo (missing "the")

* You need to decide whether or not sBTC is going to be (or going to emulate) a gas asset before taking about it here.

- **Programmability:** [Clarity](https://docs.stacks.co/clarity/overview) is the smart contract language on Stacks, which allows developers to encode essential business logic on a blockchain. Using smart contracts, developers can build more expressive decentralized applications that interact with sBTC, such as DeFi protocols, stablecoins, payments and more.
- **Fast Blocks:** The Stacks Nakamoto Upgrade, proposed in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md#proposed-solution), enables fast blocks where “user-submitted transactions will now take on the order of seconds, instead of tens of minutes.” Thus, sBTC on Stacks Nakamoto will offer an improvement to Bitcoin’s current transaction times. 

> The sBTC protocol not only addresses the limitations of the Bitcoin scripting system but also provides a secure and decentralized solution for utilizing Bitcoin in various applications.

Reviewer 3: You have mentioned "secure" several times by now.  I think you should quantify what that means earlier (just a sentence will do).

## Design

> This proposal describes a system in which the following are true:

Reviewer 3: Drop this in favor of:

> While the exact design of the first sBTC implementation is being worked on, the wrapped nature of the sBTC token means that any such system would have the following properties:

> - sBTC is a SIP-10 token backed by 1:1 by BTC.

Reviewer 3: "SIP-010", not "SIP-10"

- The sBTC peg wallet is maintained by the set of sBTC signers. These signers are responsible for the security and maintenance of the wallet, ensuring that sBTC is redeemable for BTC.
- Bitcoin can be converted into sBTC within 3 Bitcoin blocks.

> - sBTC can be converted into Bitcoin within 6 Bitcoin blocks. This ensures sBTC on and off ramps are faster than other BTC assets on the market.

Reviewer 3: Wait, are you sure they're faster?  What's the current fastest, and what stops them from just reducing their confirmation count?  Also, if the reason their confirmation counts are higher is due to improving security, then what is it about this system that makes fewer confirmations acceptable?

> - The sBTC SIP-10 token contract remains consistent across sBTC releases. This provides reliability for users and developers, meaning no adjustment from builders will be needed as the system evolves through subsequent sBTC releases.  

Reviewer 3: I don't think this is a given.

Reviewer 3: That is, I think you want to leave open the possibility that the sBTC contract API and implementation can change.  We don't know what the future will hold, so we don't want to nail ourselves down to a commitment we can't (or don't want to) uphold.

> ## Overview of the sBTC Bootstrap Release

Reviewer 3: This section and the next section strongly imply the existence of a separate sBTC implementation.  I don't think this SIP should talk about this at all, since we don't want to commit to that so openly.  Instead, this and the next section should instead simply enumerate all of the signer responsibilities that will be true regardless of what the sBTC implementation looks like.  Namely:

* Describing the responsibilities of the sBTC signers
   * They react to requests to deposit BTC
   * They react to requests to withdraw BTC
   * The stay online often enough that deposits and withdraws happen in a timely manner
   * They have to keep their signer hosts secure, so their keys don't get stolen
   * They have to proactively re-key every so often, which means moving the BTC to a new UTXO
* Signers must be independent entities 
* Signers must consolidate BTC as it is deposited
   * They must somehow ensure that that transaction fee is paid for (e.g. the user can specify the max fee, as you described in your design document)
* Signers must deduct transaction fees from users in order to fund BTC withdrawal transactions
   * They must somehow ensure that the transaction fee is paid for (e.g. they deduct it from the user, and they set a minimum sBTC withdrawal amount)
* Signers must coordinate to set and advertise the fee parameters of the system
   * They must decide a minimum sBTC peg-out
   * They must decide a STX transaction fee for minting the sBTC (who pays for this, btw?  If it's the signers, then this document must state that the signers must have a STX balance to pay it)

Once these are all listed, you should then use it to determine the criteria for selecting a suitable signer.

In the bootstrap phase, the criteria for sBTC Signers is determined through the community governance process of ratifying this SIP. sBTC Signers will be responsible for signing sBTC deposit and withdrawal transactions on the network. During this phase, sBTC will have an unchangeable and distinct signer set that will not explicitly be part of Stacks consensus. As a result, this release can activate without a hard fork.

> Management of the sBTC peg wallet on the Bitcoin blockchain is decentralized, involving the sBTC signer set rather than a single custodian. This ensures a more resilient and trustworthy system, where signers are economically incentivized to execute peg-out transactions efficiently. The system is live ("resilient") if at least 70% of the sBTC signer voting power are online and honest. Then (and only then), deposits and withdrawals happen in a timely manner. The system is safe ("trustworthy") if at least 30% of the sBTC signer voting power is honest. Then, no theft of funds can occur.

Reviewer 3: It doesn't "ensure" anything; it "enables" a more resilient and trustworthy system.

Also, how are the signers economically incentivized to be honest?

Reviewer 3: Asking the latter because you disavow slashing in the next section. That's going to be a surprise for many readers, who are used to thinking about incentivization arising out of giving token rewards for good behavior and slashing for bad behavior (and neither happen here).

> ### Differences in the sBTC Bootstrap Phase:

Reviewer 3: Recommend dropping, since this SIP only really needs to talk about the boostrap phase (so it doesn't matter how it differs from an undescribed, unplanned, future system)

- Eligibility criteria to become an sBTC Signer will be selected through an open community governance process. The eligibility criteria to become an sBTC signer is described below and will take into account Signer performance and availability. 
- sBTC Bootstrap Signers are separate from Stacks Signers. The bootstrapping period uses a separate subset of Stacks signers to secure the sBTC protocol and Signer operations are not explicitly linked with slashing of rewards.
- sBTC deposits will be triggered via an API call. During the bootstrap phase sBTC deposit fulfillment must be initiated via an API call to alert the signers to the presence of a deposit. 

> ### Auxiliary Features

Reviewer 3: This belongs as an appendix

Auxiliary features of the sBTC Bootstrap protocol are described below. 

> - **Stacks Transaction Fee Sponsorship:** sBTC will include the option to have sBTC transactions on Stacks be sponsored in return for some sBTC. Using the approach suggested in this [issue](https://github.com/stacks-network/stacks-core/issues/4235), sBTC users will be able to nearly spend sBTC as gas by getting support from an existing STX holder.

Reviewer: what is meant here by "nearly"?
```
sBTC users will be able to nearly spend sBTC as gas by getting support from an existing STX holder.
```

Author: That means the user will be able to pay for their transaction fees with sBTC, but it will not be supported at the protocol level. It will be supported at the wallet level via sponsored txs.

Reviewer: ahh i see now. i still don't like the wording, but i can't think of any alternatives to make it more clear. resolving.

> - **Stacks Transaction Fee Sponsorship:** sBTC will include the option to have sBTC transactions on Stacks be sponsored in return for some sBTC. Using the approach suggested in this [issue](https://github.com/stacks-network/stacks-core/issues/4235), sBTC users will be able to nearly spend sBTC as gas by getting support from an existing STX holder.

Reviewer 3: Instead of linking the issue, can you put a summary here?

Reviewer 3: I agree with @wileyj that the wording about "nearly" is hard to parse.  How about this?

`sBTC users will be able to pay for their transaction fees in sBTC with support from an existing STX holder, provided the wallet supports it.`

> - **Signer Key Rotation:** Mechanisms are provided for the scenario where a signer wants to rotate their key. For this to happen, signers must coordinate offline and vote on-chain on the new signer set (aka set of keys). Once the new signer set is determined, the signers conduct a wallet handoff and re-execute DKG.

Reviewer 3: DKG is not defined yet

Reviewer 3: Also, DKG is prescriptive of an implementation, so I'd strongly recommend not using it.

> ## sBTC Bootstrap Signers

Reviewer 3: Drop this section per the above advice

### Responsibilities

> The sBTC bootstrap signers are responsible for accepting or rejecting all sBTC operations submitted, and for a transaction to be fulfilled at least 70% of the signers need to approve the fulfilling of the transaction; this means that the liveness and reliability of the signers is crucial to the success of the protocol.

Reviewer 3: This is a run-on sentence

> While up to 30% of the signers can be down without a user impact to the functioning of the protocol, it becomes more critical for the rest of the signers to approve sBTC operations because operations necessarily still need to meet 70% of the original signing power. If more than 30% of signers become unavailable no sBTC operations will be approved because it will be impossible to get 70% approval when less than 70% are online. 

Reviewer: ```suggestion
While up to 30% of the signers can be down without any user impact to the functioning of the protocol, it becomes more critical for the rest of the signers to approve sBTC operations because operations necessarily still need to meet 70% of the original signing power. If more than 30% of signers become unavailable, no sBTC operations will be approved because it will be impossible to get the required 70% approval when less than 70% are online. 
```

> While up to 30% of the signers can be down without any user impact to the functioning of the protocol, it becomes more critical for the rest of the signers to approve sBTC operations because operations necessarily still need to meet 70% of the original signing power. If more than 30% of signers become unavailable, no sBTC operations will be approved because it will be impossible to get the required 70% approval when less than 70% are online. 

Reviewer 3: "offline" instead of "down"

> An operation that isn’t approved will become spendable by the user without bridging to the other blockchain after a period of time without Signer interaction.

Reviewer 3: "will become spendable by the user after a predetermined timeout has elapsed, measured in Bitcoin blocks."

### Eligibility Criteria
For the sBTC Bootstrap release Signers will run the sBTC binary in addition to the core Stacks signer software and must meet the following criteria in order to strongly ensure reliable functioning of the sBTC protocol at all times.

> The following eligibility criteria has been used to identify the sBTC Bootstrap Signers:  

Reviewer 3: "has been"? When was this decision made?

- **Stacks 2.5 Participation:** Active participation running a signer on Stacks 2.5 testnet or mainnet. 
- **Technical Performance and Uptime:** Consistency in operational status and network stability. 
- **Communication & Availability:** Running an sBTC Signer is a pivotal role in the Stacks ecosystem, which requires a high degree of availability and communication with Stacks core developers. Signers should generally be able to respond to updates within 24 hours. 

> - **Ecosystem Alignment:** Commitment to the growth of the Stacks ecosystem with contributions to support the network. Examples include (but are not limited to): publishing independent research or contributing to a Stacks working group.

Reviewer 3: I have the same questions as before -- how do you measure these in an unambiguous, falsifiable manner?

I think this is of paramount importance for building trust in these signers.  How do other users determine that the selected signers actually fit these criteria?  I don't think they should just take our word for it, because for all they know, we could be playing favorites or running sockpuppets.  Let me put it this way -- how does a 3rd party (not us, not the voters) determine that a particular signer has meet these criteria? 

The SC will also need to know these criteria, because they'll need to evaluate signers with them in order to determine whether or not the SIP's activation terms have been met.

### Selection Process

> The sBTC Bootstrap Signers will be selected from the group of eligible signers by the [sBTC working group](https://github.com/orgs/stacks-network/discussions/469).

Reviewer 3: Nit -- you're not consistent in your capitalization here.  Is it "sBTC Bootstrap Signers," "sBTC bootstrap Signers," or "sBTC Bootstrap signers"?

> ## Comparison to Other Protocols

Reviewer 3: Per SIP-000, this is "Related Work."

### [tBTC v2](https://whitepaper.io/document/691/tbtc-whitepaper)
tBTC is an open membership system, where the BTC is managed by a rotating set of randomly selected nodes which manage a threshold wallet. The system requires that  51-of-100 randomly selected wallet signers must collaborate to produce a proper signature. 

### [RBTC](https://rootstock.io/static/a79b27d4889409602174df4710102056/RS-whitepaper.pdf)
Rootstock’s (RSK) 2-way peg protocol is called “the Powpeg”. Peg operations settle to Bitcoin via merge mining on the RSK side-chain. Peg operators are incentivized by earning a portion of Rootstock transaction fees. PowPeg operators keep specialized hardware called PowHSMs active and connected to special types of Rootstock full nodes. Since the Bitcoin blockchain and the Rootstock sidechain are not entangled in a single blockchain or in a parent-child relation, peg-in and peg-out transactions require a high number of block confirmations. Peg-ins require 100 Bitcoin blocks, and peg-outs require 200 Bitcoin blocks.

## Activation

> sBTC Bootstrap is designed to activate on Stacks Nakamoto as defined in [SIP-021](https://github.com/stacksgov/sips/blob/feat/sip-021-nakamoto/sips/sip-021/sip-021-nakamoto.md). Therefore, this SIP is only meaningful when SIP-021 activates. The sBTC Working Group plans to observe at least 2-4 weeks of network behavior on Stacks Nakamoto to ensure a stable release. After this period, sBTC Bootstrap can be activated on the Stacks network without requiring a separate hard fork.

Reviewer 3: This activation section is unacceptable for several reasons.

* This usurps the SC's authority to activate SIPs and places it into an informal body outside the SIP process. That is never going to be permitted.  The sBTC Working Group should not be part of the decision-making process at all.

* There is no way for the SC to vet the list of proposed signers.  They are expected to blindly accept a list from people outside the SIP process.  The criteria for accepting a signer (described above) are ambiguous and hard to independently validate.  This prevents the SC members from acting as stewards of the blockchain; it instead makes them the fall-people in case something goes wrong.

* There is no contingency plan in case no signers or not enough signers can be found.  Furthermore, there is no threshold for the minimum number of signers that must be found, nor is there any description of how their voting power will be distributed.  For example, a list of 100 signers where 1 signer has 70% of the voting power would be an unacceptable outcome.

* There is no prescribed way to involve _sBTC stakeholders_ in the decision as to who the signers will be.  Why should prospective sBTC users trust these signers?

* The activation of this SIP is not tied to user traction, which is historically a criterion we have used for application-layer SIPs.  I think this SIP should be considered rejected if it fails to gain enough BTC locked up within a given window of time.

* I'm not sure that a vote from STX holders is enough (or even necessary, although nice-to-have in any case).  I thought a big part of this SIP's value-add was that it would enable BTC users _who don't want to use STX_ to use the Stacks blockchain?  If so, then I think this SIP's activation needs to be contingent on some amount of BTC _pledged_ to this system, since that would indicate that there's interest from this body of users (who wouldn't be counted in a STX-only coin vote).

### Process of Activation
Users can vote to approve this SIP with either their locked/stacked STX or with unlocked/liquid STX, or both. The criteria for the stacker and non-stacker voting is as follows.

**For Stackers:**
In order for this SIP to activate, the following criteria must be met by the set of Stacked STX:
- At least 80 million Stacked STX must vote at all to activate this SIP.
- Of the Stacked STX that vote, at least 66% of them must vote "yes."

The voting addresses will be;
- **Bitcoin Yes Address:** 3Jq9UT81fnT2t24XjNVY7wijpsSmNSivbK
- **Bitcoin No Address:** 3QGZ1fDa97yZCXpAnXQd6JHF4CBC6bk1r4
- **Stacks Yes Address:** SP36GHEPEZPGD53G2F29P5NEY884DXQR7TX90QE3T
- **Stacks No Address:** SP3YAKFMGWSSATYNCKXKJHE2Z5JJ6DH88E4T8XJPK

which encode the hashes of the following phrases into bitcoin / stacks addresses:

- Yes to A Decentralized Two-Way Bitcoin Peg

> - No to A Decentralized Two-Way Bitcoin Peg

Reviewer 3: Describe how

Reviewer 5: Following has been used @jcnelson where proposalId is the string to encode.
```
export async function generateAddresses(proposalId:string) {

  console.log('generateAddresses: ' + proposalId)
  let encoder = new TextEncoder();
  let encoded = encoder.encode(`Yes to ${proposalId}`);
  let hash256 = sha256(encoded);
  const hash160Y = ripemd160(hash256);

  encoder = new TextEncoder();
  encoded = encoder.encode(`No to ${proposalId}`);
  hash256 = sha256(encoded);
  const hash160N = ripemd160(hash256);

  const net = getNet(getConfig().network);
  
  const p2shObjY = btc.p2sh({type:'sh', script: btc.Script.encode(['DUP', 'HASH160', hash160Y, 'EQUALVERIFY', 'CHECKSIG'])}, net)
  const p2shObjN = btc.p2sh({type:'sh', script: btc.Script.encode(['DUP', 'HASH160', hash160N, 'EQUALVERIFY', 'CHECKSIG'])}, net)

  console.log(p2shObjY.address)
  console.log(p2shObjN.address)

  const netPrefix = (getConfig().network === 'testnet') ? 26 : 22;
  const yStxAddress = c32address(netPrefix, hex.encode(hash160Y))
  const nStxAddress = c32address(netPrefix, hex.encode(hash160N))

  const votingAddresses = {
    yBtcAddress: p2shObjY.address as string,
    nBtcAddress: p2shObjN.address as string,
    yStxAddress,
    nStxAddress,
  }
  console.log('generateAddresses: ', votingAddresses)
  return votingAddresses;
}
```

Stackers (pool and solo) vote by sending a stacks dust transaction to the corresponding stacks address from the account where their stacks are locked.

> Solo stackers only, can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.

Reviewer: ```suggestion
Solo stackers only can also vote by sending a bitcoin dust transaction (6000 sats) to the corresponding bitcoin address.
```

**For Non-Stackers:**
Users with liquid STX can vote on proposals using the Ecosystem DAO. Liquid STX is the users balance, less any STX they have locked in PoX stacking protocol, at the block height at which the voting started (preventing the same STX from being transferred between accounts and used to effectively double vote). This is referred to generally as "snapshot" voting.

For this SIP to pass, 66% of all liquid STX committed by voting must be in favor of the proposal. This precedent was set by [SIP-015](https://github.com/stacksgov/sips/blob/feat/sip-015/sips/sip-015/sip-015-network-upgrade.md). 

The act of not voting is the act of siding with the outcome, whatever it may be. We believe that these thresholds are sufficient to demonstrate interest from Stackers -- Stacks users who have a long-term interest in the Stacks blockchain's successful operation -- in performing this upgrade.

## Appendix

### Specification

> **Deposits**

Reviewer 3: This needs to be fleshed out
